code/modules/antagonists/nightmare/nightmare_organs.dm

```diff
@@ -0,0 +1,101 @@
+/// How many life ticks are required for the nightmare's heart to revive the nightmare.
+#define HEART_RESPAWN_THRESHHOLD 40
+/// A special flag value used to make a nightmare heart not grant a light eater. Appears to be unused.
+#define HEART_SPECIAL_SHADOWIFY 2
+
+
+/obj/item/organ/brain/nightmare
+	name = "tumorous mass"
+	desc = "A fleshy growth that was dug out of the skull of a Nightmare."
+	icon_state = "brain-x-d"
+	var/obj/effect/proc_holder/spell/targeted/shadowwalk/shadowwalk
+
+/obj/item/organ/brain/nightmare/Insert(mob/living/carbon/M, special = 0)
+	..()
+	if(M.dna.species.id != "nightmare")
+		M.set_species(/datum/species/shadow/nightmare)
+		visible_message("<span class='warning'>[M] thrashes as [src] takes root in [M.p_their()] body!</span>")
+	var/obj/effect/proc_holder/spell/targeted/shadowwalk/SW = new
+	M.AddSpell(SW)
+	shadowwalk = SW
+
+
+/obj/item/organ/brain/nightmare/Remove(mob/living/carbon/M, special = 0)
+	if(shadowwalk)
+		M.RemoveSpell(shadowwalk)
+	..()
```

Comment:
```suggestion
	return ..()
```

Reply:
It was like that originally and I don't want to break something by changing it. I'll take a look.

Reply:
Done